app/models/power_relay.rb

```diff
@@ -52,10 +52,19 @@ def activate_secondary_outlet
   # boolean - The current on/off status of the outlet (and secondary outlet, if configured).
   def query_status
     log_power_relay_connection(:status) do
-      relay_status = relay_connection.status(outlet)
+      relay_status = nil
+      max_retries = 2
+
+      retry_on_error(max_retries, NetBooter::Error) do
+        relay_status = relay_connection.status(outlet)
+      end
+
       if secondary_outlet
-        secondary_outlet_status = relay_connection.status(secondary_outlet)
+        retry_on_error(max_retries, NetBooter::Error) do
+          relay_connection.status(secondary_outlet)
```

Comment:
Does this method return the actual status of the outlet? If so, I don't think it's being used. Maybe you need to assign it to a variable and add it in the method's output. Does that make any sense?

Reply:
Before we weren't returning it. I think the idea is that both outlets should have the same status, so I'm not even sure why we read both of them. 

I could reach out to the clients and confirm the expected behavior. However, they haven't brought up having an issue with the current implementation.

Reply:
I think we can remove that call then as it's not used